Source/WTF/wtf/Dominators.h

```diff
@@ -294,6 +314,129 @@ class Dominators {
+        WTF_MAKE_FAST_ALLOCATED;
+        constexpr static uint16_t undefinedIdom = std::numeric_limits<uint16_t>::max();
+    public:
+        IterativeDominance(Graph& graph)
+            : m_graph(graph)
+        {
+            // We only use this for small-ish graphs. So, we exploit that to use
+            // smaller integers for idom information. We mostly use uint16_t for
+            // our analysis, but we exploit int16_t when computing reverse
+            // postorder. We expect Lengauer-Tarjan to beat us beyond a few ten
+            // thousand blocks anyway so this should be fine.
+            RELEASE_ASSERT(graph.numNodes() < std::numeric_limits<int16_t>::max());
+            m_idoms.fill(undefinedIdom, graph.numNodes());
+        }
+
+        void computeReversePostorder()
+        {
+            BitSet<std::numeric_limits<int16_t>::max()> visited;
+            visited.clearAll();
+
+            Vector<int16_t> workList;
+            int16_t rootIndex = m_graph.index(m_graph.root());
+            workList.append(rootIndex);
+            visited.set(rootIndex);
+
+            while (workList.size()) {
+                int16_t index = workList.takeLast();
+                if (index < 0) {
+                    // Negative indices mark nodes we're revisiting.
+                    m_order.append(~index);
+                    continue;
+                }
+                const auto& successors = m_graph.successors(m_graph.node(index));
+                if (!successors.size()) {
+                    m_order.append(index);
+                    continue;
+                }
+                workList.append(~index); // Revisit this index later to add it after visiting all successors.
+                for (unsigned i = 0; i < successors.size(); i ++) {
+                    int16_t index = m_graph.index(successors[i]);
+                    if (!visited.get(index)) {
+                        visited.set(index);
+                        workList.append(index);
+                    }
+                }
+            }
+            m_postorderNumbers.fill(0, m_graph.numNodes());
+            for (unsigned i = 0; i < m_order.size(); i ++)
+                m_postorderNumbers[m_order[i]] = i;
+            std::reverse(m_order.begin(), m_order.end());
+        }
+
+        uint16_t intersect(uint16_t a, uint16_t b)
+        {
+            while (a != b) {
+                while (m_postorderNumbers[a] < m_postorderNumbers[b])
+                    a = m_idoms[a];
+                while (m_postorderNumbers[b] < m_postorderNumbers[a])
+                    b = m_idoms[b];
+            }
+            return a;
+        }
+
+        void compute()
+        {
+            computeReversePostorder();
+
+            bool fixpoint = false;
+            int16_t rootIndex = m_graph.index(m_graph.root());
+            m_idoms[rootIndex] = rootIndex;
+            ASSERT(m_order[0] == rootIndex);
+            while (!fixpoint) {
+                fixpoint = true;
```

Comment:
We usually use `changed` boolean for fixpoint computation.

Reply:
Sure, I can rename it.